test/e2e/apigatewayv2/tests/test_api.py

```diff
@@ -0,0 +1,292 @@
+                                     StatementId='apigatewayv2-authorizer-invoke-permissions',
+                                     Action='lambda:InvokeFunction',
+                                     Principal='apigateway.amazonaws.com',
+                                     SourceArn=authorizer_arn)
+
+    @pytest.mark.run(order=9)
+    def test_create_route(self, apigw_validator, test_resource_values):
+        # Create an integration for the HTTP Api
+        route_ref = helper.get_route_ref(route_resource_name=test_resource_values['ROUTE_NAME'])
+        route_resource_data = load_resource_file(
+            SERVICE_NAME,
+            "route",
+            additional_replacements=test_resource_values
+        )
+        logging.error(f"route resource: {route_resource_data}")
+
+        # Create the k8s integration resource
+        k8s.create_custom_resource(route_ref, route_resource_data)
+        cr = k8s.wait_resource_consumed_by_controller(route_ref)
+        assert cr is not None
+        assert k8s.get_resource_exists(route_ref)
+        route_id = cr['status']['routeID']
+        test_resource_values['ROUTE_ID'] = route_id
+        apigw_validator.assert_route_is_present(api_id=test_resource_values['API_ID'], route_id=route_id)
+
+    @pytest.mark.run(order=10)
+    def test_create_stage(self, apigw_validator, test_resource_values):
+        # Create an integration for the HTTP Api
+        stage_ref = helper.get_stage_ref(stage_resource_name=test_resource_values['STAGE_NAME'])
+        stage_resource_data = load_resource_file(
+            SERVICE_NAME,
+            "stage",
+            additional_replacements=test_resource_values
+        )
+        logging.error(f"stage resource: {stage_resource_data}")
+
+        # Create the k8s integration resource
+        k8s.create_custom_resource(stage_ref, stage_resource_data)
+        cr = k8s.wait_resource_consumed_by_controller(stage_ref)
+        assert cr is not None
+        assert k8s.get_resource_exists(stage_ref)
+        apigw_validator.assert_stage_is_present(
+            api_id=test_resource_values['API_ID'],
+            stage_name=test_resource_values['STAGE_NAME']
+        )
+
+    @pytest.mark.run(order=11)
+    def test_perform_invocation(self, apigw_validator, test_resource_values):
+        time.sleep(DEPLOYMENT_WAIT_AFTER_SECONDS)
+        api_ref = helper.get_api_ref(api_resource_name=test_resource_values['API_NAME'])
+        api_endpoint = helper.get_api_endpoint(api_ref=api_ref)
+        invoke_url = "{api_endpoint}/{stage_name}/{route_key}"\
+            .format(api_endpoint=api_endpoint, stage_name=test_resource_values['STAGE_NAME'],
+                    route_key=test_resource_values['ROUTE_KEY']
+                    )
+        response = requests.request(method='GET', url=invoke_url, headers={'Authorization': 'SecretToken'})
+        assert 200 == response.status_code
+
+    @pytest.mark.run(order=-5)
```

Comment:
Did you realize that you're using negative numbers for the delete test orders?

Reply:
This is a genius way of telling pytest to run it in an order offset from the end of the tests. Matching the test that runs with `order=1` with its deletion test with `order=-1`, etc.